dig_test.go

```diff
@@ -627,6 +629,69 @@ func TestEndToEndSuccess(t *testing.T) {
 		}), "both objects should be successfully resolved on Invoke")
 	})
 
+	t.Run("struct constructor with as interface option", func(t *testing.T) {
+		c := New()
+
+		provider := c.Provide(
+			func() *bytes.Buffer {
+				var buf bytes.Buffer
+				buf.WriteString("foo")
+				return &buf
+			},
+			As(new(fmt.Stringer), new(io.Reader)),
+		)
+
+		require.NoError(t, provider, "provide failed")
```

Comment:
`provider` appears to be an error. it should probably be named `err`.
```suggestion
		require.NoError(t, err, "provide failed")
```